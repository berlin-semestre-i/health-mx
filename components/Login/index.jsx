import React, { PureComponent } from 'react' 
import { Grid, Form, Button, Container, Image } from 'semantic-ui-react'
import Link from 'next/link'
import styled from 'styled-components'
import media from 'styled-media-query'

const CustomGrid = styled(Grid)`
  && {
    height: 100vh;
    margin: 0;
  }

  ${media.greaterThan('medium')`
    && .nine.column{
      background-image: url("../../static/assets/enfermera.png");
      background-repeat: no-repeat;
      background-size: 100%;
    }
  `}

  ${media.lessThan('medium')`
    && .nine.column{
      display: none;
    }
  `}
  
`
const RightColumn = styled(Grid.Column)`
  ${media.lessThan('large')`
    margin: 2rem auto auto auto;
  `}

  ${media.greaterThan('large')`
    margin: auto;
  `}
`
const LoginContainer = styled(Container)`

  ${media.lessThan('large')`
    padding: 0 1rem;
  `}

  ${media.greaterThan('large')`
    padding: 0 4.5rem;
  `}

  font-family: -apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,'Helvetica Neue',Arial,Helvetica,sans-serif;

  &&& .field, &&& .button {
    margin-bottom: 2em;
  }

  &&& .field label {
    font-size: 14px;
    font-weight: normal;
  }

  &&& .field label:after {
    display: none;
  }

  &&& img {
    max-width: 65%;
    margin-bottom: 3rem;
  }
`
const SecondaryContent = styled.div`
  & {
    text-align: center;
    font-size: 13px;
    line-height: 1.5rem;
    color: #3d5170;
  }

  &.footer {
    margin-top: 5rem;
  }

  & a {
    color: #007bff;
    font-weight: 500;
    text-decoration: none;
  }
`

const LoginButton = styled(Button)`
  width: 100%;
`

class Login extends PureComponent {
  render() {
    return(
      <CustomGrid>
        <Grid.Column width={9}/>
        <RightColumn mobile={16} tablet={7} computer={7}>
          <LoginContainer>
            <Image src="../../static/assets/login-logo.png" />  
            <Form>
              <Form.Input
                label="NSS o correo electrónico"
                placeholder="NSS o jorge@ejemplo.com" 
                required
              />
              <Form.Input
                label="Contraseña"
                placeholder="••••••••••" 
                type="password"
                required
              />
              <LoginButton small="true" primary type="submit">Ingresar</LoginButton>
            </Form>
            <SecondaryContent>
              ¿Aún no tienes tu cuenta?
              <Link as={`/registro`} href={`/register`}>
                <a> Regístrate</a>
              </Link>
            </SecondaryContent>
            <SecondaryContent>
              <Link as={`/recuperar`} href={`/recovery`}>
                <a>Olvidé mi contraseña</a>
              </Link>
            </SecondaryContent>
            <SecondaryContent className="footer">
              © Instituto Mexicano del Seguro Social. Todos los derechos reservados.
            </SecondaryContent>
          </LoginContainer>
        </RightColumn>
      </CustomGrid>
    )
  }
}

export default Login