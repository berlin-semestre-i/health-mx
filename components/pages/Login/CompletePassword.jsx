import React, { PureComponent } from 'react'
import { Form, Button, Grid } from 'semantic-ui-react'
import AuthenticationLayout from '../../Layout/AuthenticationLayout'
import styled from 'styled-components'
import { Auth } from 'aws-amplify'
import { getCookie, removeCookie, setCookie } from '../../../utils/session'
import { redirectIfAuthenticated } from '../../../utils/auth'

class CompletePassword extends PureComponent {

  state = {
    username: '',
    name: '',
    familyName: '',
    phoneNumber:
      getCookie('cognito-phone-number')
        ? getCookie('cognito-phone-number')
        : ''
    ,
    email:
      getCookie('cognito-email')
        ? getCookie('cognito-email')
        : ''
    ,
    oldPassword: '',
    newPassword: '',
  }

  componentDidMount() {
    const username = getCookie('cognito-username')
    this.setState({
      username: username ? username : '',
    })
  }

  onChange = (e, { name, value }) => {
    this.setState({ [[name]]: value })
  }

  completePassword = async () => {
    const {
      name,
      familyName,
      phoneNumber,
      email,
      username,
      oldPassword,
      newPassword,
    } = this.state

    let attributes = {
      name: name,
      'family_name': familyName,
    }

    if (phoneNumber || phoneNumber !== '') {
      attributes['phone_number'] = `+52${phoneNumber}`
    }
    if (email || email !== '') {
      attributes.email = email
    }

    try {
      const user = await Auth.signIn(username, oldPassword)
      await Auth.completeNewPassword(user, newPassword, attributes)

      const session = await Auth.currentSession()
      setCookie('cognito-access-token', session.getAccessToken().getJwtToken())
      removeCookie('cognito-username')
      removeCookie('cognito-phone-number')
      removeCookie('cognito-email')
      redirectIfAuthenticated()
    } catch(err) {
      console.log(err)
    }
  }

  render() {
    const {
      name,
      familyName,
      phoneNumber,
      email,
      oldPassword,
      newPassword,
    } = this.state

    return (
      <AuthenticationLayout>
        <Form>
          <Grid>
            <Grid.Column width={8}>
              <Form.Input
                name="name"
                label="Nombre"
                icon="user"
                iconPosition="left"
                value={name}
                onChange={this.onChange}
                placeholder="Morty"
                required
              />
            </Grid.Column>
            <Grid.Column width={8}>
              <Form.Input
                name="familyName"
                label="Apellido"
                icon="user"
                iconPosition="left"
                value={familyName}
                onChange={this.onChange}
                placeholder="Smith"
                required
              />
            </Grid.Column>
            <Grid.Column width={6}>
              <Form.Input
                name="phoneNumber"
                label="Teléfono Celular"
                icon="phone"
                iconPosition="left"
                value={phoneNumber}
                onChange={this.onChange}
                placeholder="3331112222"
              />
            </Grid.Column>
            <Grid.Column width={10}>
              <Form.Input
                name="email"
                label="Correo Electrónico"
                icon="at"
                iconPosition="left"
                value={email}
                onChange={this.onChange}
                placeholder="morty.smith@email.com"
              />
            </Grid.Column>
            <Grid.Column width={16}>
              <Form.Input
                name="oldPassword"
                label="Contraseña Vieja"
                icon="lock"
                iconPosition="left"
                value={oldPassword}
                onChange={this.onChange}
                placeholder="••••••••••"
                autoComplete="off"
                type="password"
                required
              />
            </Grid.Column>
            <Grid.Column width={16}>
              <Form.Input
                name="newPassword"
                label="Contraseña Nueva"
                icon="lock"
                iconPosition="left"
                value={newPassword}
                onChange={this.onChange}
                placeholder="••••••••••"
                autoComplete="off"
                type="password"
                required
              />
            </Grid.Column>
          </Grid>
          <CompleteButton
            small="true"
            primary
            type="submit"
            onClick={this.completePassword}
          >
            Ingresar
          </CompleteButton>
        </Form>
      </AuthenticationLayout>
    )
  }

}

export default CompletePassword

const CompleteButton = styled(Button)`
  & {
    width: 100%;
  }
`
