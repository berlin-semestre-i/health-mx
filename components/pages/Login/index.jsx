import React, { PureComponent } from 'react'
import { Form, Button, Image } from 'semantic-ui-react'
import AuthenticationLayout from '../../Layout/AuthenticationLayout'
import Link from 'next/link'
import styled from 'styled-components'
import { Auth } from 'aws-amplify'
import redirect from '../../../utils/redirect'
import { setCookie, removeCookie } from '../../../utils/session'
import { redirectIfAuthenticated } from '../../../utils/auth'

class Login extends PureComponent {

  state = {
    username: '',
    password: '',
  }

  onChange = (e, { name, value }) => {
    this.setState({ [[name]]: value })
  }

  signIn = async () => {
    try {
      const user = await Auth.signIn(this.state.username, this.state.password)
      if (user.challengeName === 'NEW_PASSWORD_REQUIRED') {
        setCookie('cognito-username', this.state.username)
        redirect('/auth/complete-password')
      }
      else {
        removeCookie('cognito-username')
        const session = await Auth.currentSession()
        setCookie('cognito-access-token', session.getAccessToken().getJwtToken())
        redirectIfAuthenticated()
      }
    } catch (err) {
      console.log(err)
    }
  }

  render() {
    const { username, password } = this.state
    return (
      <AuthenticationLayout>
        <Image src="../../static/images/login-logo.png" />
        <Form>
          <Form.Input
            label="NSS o correo electrónico"
            icon="user"
            value={username}
            name="username"
            onChange={this.onChange}
            iconPosition="left"
            placeholder="NSS o jorge@ejemplo.com"
            required
          />
          <Form.Input
            label="Contraseña"
            icon="lock"
            value={password}
            onChange={this.onChange}
            iconPosition="left"
            placeholder="••••••••••"
            autoComplete="off"
            name="password"
            type="password"
            required
          />
          <LoginButton
            small="true"
            primary
            type="submit"
            onClick={this.signIn}
          >
            Ingresar
          </LoginButton>
        </Form>
        <InfoContent>
          ¿Aún no tienes tu cuenta?
          <Link href="/register">
            <a> Regístrate</a>
          </Link>
        </InfoContent>
        <InfoContent>
          <Link href="/recovery">
            <a>Olvidé mi contraseña</a>
          </Link>
        </InfoContent>
        <InfoContent className="footer">
                    © Instituto Mexicano del Seguro Social. Todos los derechos reservados.
        </InfoContent>
      </AuthenticationLayout>
    )
  }
}

export default Login

const InfoContent = styled.div`
  & {
    text-align: center;
    font-size: 13px;
    line-height: 1.5rem;
  }

  &.footer {
    margin-top: 5rem;
  }

  & a {
    font-weight: 500;
  }
`

const LoginButton = styled(Button)`
  & {
    width: 100%;
  }
`
