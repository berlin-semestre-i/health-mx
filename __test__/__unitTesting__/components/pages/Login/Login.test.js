import React from 'react'
import { shallow } from 'enzyme'
import Login from '../../../../../components/pages/Login/index'

describe('Login', () => {
  it('renders', () => {
    const component = shallow(<Login />)
    expect(component).toMatchSnapshot()
  })
})