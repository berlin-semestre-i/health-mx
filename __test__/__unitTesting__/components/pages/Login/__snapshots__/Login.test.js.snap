// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders 1`] = `
<AuthenticationLayout>
  <Image
    as="img"
    src="../../static/images/login-logo.png"
    ui={true}
  />
  <Form
    as="form"
  >
    <FormInput
      as={[Function]}
      control={[Function]}
      icon="user"
      iconPosition="left"
      label="NSS o correo electrónico"
      name="username"
      onChange={[Function]}
      placeholder="NSS o jorge@ejemplo.com"
      required={true}
      value=""
    />
    <FormInput
      as={[Function]}
      autoComplete="off"
      control={[Function]}
      icon="lock"
      iconPosition="left"
      label="Contraseña"
      name="password"
      onChange={[Function]}
      placeholder="••••••••••"
      required={true}
      type="password"
      value=""
    />
    <Login__LoginButton
      onClick={[Function]}
      primary={true}
      small="true"
      type="submit"
    >
      Ingresar
    </Login__LoginButton>
  </Form>
  <Login__InfoContent>
    ¿Aún no tienes tu cuenta?
    <Link
      href="/register"
    >
      <a>
         Regístrate
      </a>
    </Link>
  </Login__InfoContent>
  <Login__InfoContent>
    <Link
      href="/recovery"
    >
      <a>
        Olvidé mi contraseña
      </a>
    </Link>
  </Login__InfoContent>
  <Login__InfoContent
    className="footer"
  >
    © Instituto Mexicano del Seguro Social. Todos los derechos reservados.
  </Login__InfoContent>
</AuthenticationLayout>
`;
